### PR TITLE
Fix the wrong way to cancel the context

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:
I'm sorry for my making a stupid but serious mistake with my carelessness.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nghialv** [comment]
😄 
/lgtm

File: pkg/app/piped/executor/analysis/analysis.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 				}
 				status = model.StageStatus_STAGE_SKIPPED
 				// Stop the context to cancel all running analysises.
-				ctxWithTimeout.Done()
+				cancel()
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/analysis/analysis.go:120]: as convention, should rename this stop function as `stop()`

Comment (knanao) [reply pkg/app/piped/executor/analysis/analysis.go:120]: Is this a convention for PipeCD team?
I guess `cancel()` is often used in this case.

Comment (khanhtc1202) [reply pkg/app/piped/executor/analysis/analysis.go:120]: Ah, sorry, just my tatse. For `WithCancel` will use `cancel()` and for `WithTimeout` will use `stop()`. But not a big deal, if other use the same 👀

**Comment by khanhtc1202** [comment]
Here you go
/approve

**Pull request merged**